zap/src/main/java/org/zaproxy/zap/extension/ascan/PolicyAllCategoryPanel.java

```diff
@@ -515,7 +517,9 @@ public void setScanPolicy(ScanPolicy scanPolicy) {
         this.getPolicySelector().setSelectedItem(scanPolicy.getName());
         this.setThreshold(scanPolicy.getDefaultThreshold());
         this.setStrength(scanPolicy.getDefaultStrength());
-        this.getAllCategoryTableModel().setPluginFactory(scanPolicy.getPluginFactory());
+        if (this.policy.equals(scanPolicy)) {
```

Comment:
However, I found another issue. Suppose, we update the Default Policy in Scan Policy Manager and restart ZAP, now, if we open the active scan policy panel and try to update the threshold or strength for any specific category, then changes are not reflected on the corresponding category panel. I found that for some reason, `this.policy` and `scanPolicy` do not point to the same object and hence, differ for `PolicyAllCategoryPanel` and `AllCategoryTableModel`. So I added a condition over here.

Reply:
That's all the same issue: https://github.com/zaproxy/zaproxy/issues/7716#issuecomment-1435108100

Reply:
The root cause of the issue was that the combo box was setting another policy. The latest push should fix it, please take a look.